src/features/class-rest-api.php

```diff
@@ -23,26 +23,127 @@ public function __construct() {}
+		$trending         = 'true' === $request->get_param( 'trending' ) ? true : false;
+		$tax_relation     = $request->get_param( 'tax_relation' ) ?? 'OR';
+
+		if ( ! is_string( $post_type_string ) ) {
+			$post_type_string = 'post';
+		}
+
+		/**
+		 * Filters the allowed taxonomies.
+		 *
+		 * @param array<string> $allowed_taxonomies The allowed taxonomies.
+		 * @return array<string> The allowed taxonomies.
+		 */
+		$allowed_taxonomies = apply_filters( 'wp_curate_allowed_taxonomies', [ 'category', 'post_tag' ] );
+		$taxonomies         = array_map( 'get_taxonomy', $allowed_taxonomies );
+		$taxonomies         = array_filter( $taxonomies, 'is_object' );
+		$tax_query          = [];
+		foreach ( $taxonomies as $taxonomy ) {
+			$rest_base = $taxonomy->rest_base;
+			if ( empty( $rest_base ) || ! is_string( $rest_base ) ) {
+				continue;
+			}
+			$tax_param = $request->get_param( $rest_base );
+			if ( ! is_array( $tax_param ) ) {
+				continue;
+			}
+			$terms    = isset( $tax_param['terms'] ) ? $tax_param['terms'] : [];
+			$operator = isset( $tax_param['operator'] ) ? $tax_param['operator'] : 'OR';
+			if ( empty( $terms ) ) {
+				continue;
+			}
+			$terms       = explode( ',', $terms );
+			$terms       = array_map( 'intval', $terms );
+			$terms       = array_filter( $terms, 'term_exists' ); // @phpstan-ignore-line
+			$tax_query[] = [
+				'taxonomy' => $taxonomy->name,
+				'field'    => 'term_id',
+				'terms'    => $terms,
+				'operator' => 'AND' === $operator ? 'AND' : 'IN',
+			];
+		}
+		if ( ! empty( $tax_query ) && 1 < count( $tax_query ) ) {
+			$tax_query['relation'] = $tax_relation;
+		}
+
+		$post_types = explode( ',', $post_type_string );
+		/**
+		 * Filters the allowed post types.
+		 *
+		 * @param array<string> $allowed_post_types The allowed post types.
+		 * @return array<string> The allowed post types.
+		 */
+		$allowed_post_types = apply_filters( 'wp_curate_allowed_post_types', [ 'post' ] );
+
+		$post_types = array_filter(
+			$post_types,
+			function ( $post_type ) use ( $allowed_post_types ) {
+				return in_array( $post_type, $allowed_post_types, true );
+			}
+		);
+
+		$args = [
+			'post_type'           => $post_types,
+			'posts_per_page'      => $per_page,
+			'offset'              => $offset,
+			'ignore_sticky_posts' => true,
+			'fields'              => 'ids',
+		];
+		if ( ! empty( $search_term ) ) {
+			$args['s'] = $search_term;
+		}
+		if ( ! empty( $tax_query ) ) {
+			$args['tax_query'] = $tax_query; // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
 		}
 
-		return $query_args;
+		if ( $trending ) {
+			/**
+			 * Filters the trending posts query.
+			 *
+			 * @param array<string, mixed> $posts The posts.
+			 * @param array<string, mixed> $args The arguments.
+			 */
+			$posts = apply_filters( 'wp_curate_trending_posts_query', [], $args );
+		}
+		if ( empty( $posts ) ) {
+			$query = new \WP_Query( $args );
+			$posts = $query->posts;
```

Comment:
Am I following that the "correct" results of this endpoint would be the same results that are going to appear on the frontend for the block? Would we want to use the same `Post_Queries` instance that are used when filtering the block output here instead of a plain `WP_Query`?